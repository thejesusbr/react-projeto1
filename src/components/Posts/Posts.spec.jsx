//@ts-nocheck

const { render, screen } = require('@testing-library/react');
const { Posts } = require('.');

const props = {
  posts: [
    {
      id: 1,
      title: 'Título 1',
      body: 'Corpo 1',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: 'Título 2',
      body: 'Corpo 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'Título 3',
      body: 'Corpo 3',
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render Posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /Título/i })).toHaveLength(3);

    expect(screen.getAllByRole('img', { name: /Título/i })).toHaveLength(3);

    expect(screen.getAllByText(/corpo/i)).toHaveLength(3);

    expect(screen.getByRole('img', { name: /título 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('should not render posts', () => {
    render(<Posts />);
    expect(screen.queryByRole('heading', { name: /título/i })).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
