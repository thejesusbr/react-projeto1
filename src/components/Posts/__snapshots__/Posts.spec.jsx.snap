// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Título 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        1
        : 
        Título 1
      </h2>
      <p>
        Corpo 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Título 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        2
        : 
        Título 2
      </h2>
      <p>
        Corpo 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Título 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        3
        : 
        Título 3
      </h2>
      <p>
        Corpo 3
      </p>
    </div>
  </div>
</div>
`;
